azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -96,9 +105,21 @@ def report_all_tracked():
     def poll_all_tracked():
         with CGroupsTelemetry._rlock:
             for cgroup in CGroupsTelemetry._tracked[:]:
+
                 if cgroup.name not in CGroupsTelemetry._cgroup_metrics:
                     CGroupsTelemetry._cgroup_metrics[cgroup.name] = CgroupMetrics()
-                CGroupsTelemetry._cgroup_metrics[cgroup.name].add_new_data(cgroup.controller, cgroup.collect())
+
+                metric = None
+                # noinspection PyBroadException
+                try:
+                    metric = cgroup.collect()
+                except Exception:
+                    # Wanted to have other smaller granular exceptions but the recurring nature of this call could
```

Comment:
> could  cause overfill the logs

but makes harder to debug issues when we are not collecting data

Reply:
Agreed, don't we still want to log the failure to collect metrics? It doesn't seem these failures would be transient, so we'd like to know if/why we fail.